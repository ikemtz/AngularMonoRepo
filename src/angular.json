{
  "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
  "version": 1,
  "newProjectRoot": "",
  "projects": {
    "imng-kendo-odata": {
      "projectType": "library",
      "root": "libs/imng-kendo-odata",
      "sourceRoot": "libs/imng-kendo-odata/src",
      "prefix": "imng",
      "architect": {
        "build": {
          "builder": "@nrwl/angular:package",
          "options": {
            "tsConfig": "libs/imng-kendo-odata/tsconfig.lib.json",
            "project": "libs/imng-kendo-odata/ng-package.json"
          },
          "configurations": {
            "production": {
              "tsConfig": "libs/imng-kendo-odata/tsconfig.lib.prod.json"
            }
          }
        },
        "lint": {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": {
            "tsConfig": ["libs/imng-kendo-odata/tsconfig.lib.json", "libs/imng-kendo-odata/tsconfig.spec.json"],
            "exclude": ["**/node_modules/**", "!libs/imng-kendo-odata/**"]
          }
        },
        "test": {
          "builder": "@nrwl/jest:jest",
          "options": {
            "jestConfig": "libs/imng-kendo-odata/jest.config.js",
            "tsConfig": "libs/imng-kendo-odata/tsconfig.spec.json",
            "passWithNoTests": true,
            "setupFile": "libs/imng-kendo-odata/src/test-setup.ts"
          }
        }
      },
      "schematics": {}
    },
    "imng-auth0-oidc": {
      "projectType": "library",
      "root": "libs/imng-auth0-oidc",
      "sourceRoot": "libs/imng-auth0-oidc/src",
      "architect": {
        "build": {
          "builder": "@nrwl/angular:package",
          "options": {
            "tsConfig": "libs/imng-auth0-oidc/tsconfig.lib.json",
            "project": "libs/imng-auth0-oidc/ng-package.json"
          },
          "configurations": {
            "production": {
              "tsConfig": "libs/imng-auth0-oidc/tsconfig.lib.prod.json"
            }
          }
        },
        "lint": {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": {
            "tsConfig": ["libs/imng-auth0-oidc/tsconfig.lib.json", "libs/imng-auth0-oidc/tsconfig.spec.json"],
            "exclude": ["**/node_modules/**", "!libs/imng-auth0-oidc/**"]
          }
        },
        "test": {
          "builder": "@nrwl/jest:jest",
          "options": {
            "jestConfig": "libs/imng-auth0-oidc/jest.config.js",
            "tsConfig": "libs/imng-auth0-oidc/tsconfig.spec.json",
            "setupFile": "libs/imng-auth0-oidc/src/test-setup.ts",
            "passWithNoTests": true
          }
        }
      },
      "schematics": {
        "@nrwl/angular:component": {
          "styleext": "scss"
        }
      }
    },
    "imng-kendo-grid-odata": {
      "projectType": "library",
      "root": "libs/imng-kendo-grid-odata",
      "sourceRoot": "libs/imng-kendo-grid-odata/src",
      "architect": {
        "build": {
          "builder": "@nrwl/angular:package",
          "options": {
            "tsConfig": "libs/imng-kendo-grid-odata/tsconfig.lib.json",
            "project": "libs/imng-kendo-grid-odata/ng-package.json"
          },
          "configurations": {
            "production": {
              "tsConfig": "libs/imng-kendo-grid-odata/tsconfig.lib.prod.json"
            }
          }
        },
        "lint": {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": {
            "tsConfig": [
              "libs/imng-kendo-grid-odata/tsconfig.lib.json",
              "libs/imng-kendo-grid-odata/tsconfig.spec.json"
            ],
            "exclude": ["**/node_modules/**", "!libs/imng-kendo-grid-odata/**"]
          }
        },
        "test": {
          "builder": "@nrwl/jest:jest",
          "options": {
            "jestConfig": "libs/imng-kendo-grid-odata/jest.config.js",
            "tsConfig": "libs/imng-kendo-grid-odata/tsconfig.spec.json",
            "setupFile": "libs/imng-kendo-grid-odata/src/test-setup.ts",
            "passWithNoTests": true
          }
        }
      },
      "schematics": {
        "@nrwl/angular:component": {
          "styleext": "scss"
        }
      }
    },
    "imng-application-insights-ngrx": {
      "projectType": "library",
      "root": "libs/imng-application-insights-ngrx",
      "sourceRoot": "libs/imng-application-insights-ngrx/src",
      "architect": {
        "build": {
          "builder": "@nrwl/angular:package",
          "options": {
            "tsConfig": "libs/imng-application-insights-ngrx/tsconfig.lib.json",
            "project": "libs/imng-application-insights-ngrx/ng-package.json"
          },
          "configurations": {
            "production": {
              "tsConfig": "libs/imng-application-insights-ngrx/tsconfig.lib.prod.json"
            }
          }
        },
        "lint": {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": {
            "tsConfig": [
              "libs/imng-application-insights-ngrx/tsconfig.lib.json",
              "libs/imng-application-insights-ngrx/tsconfig.spec.json"
            ],
            "exclude": ["**/node_modules/**", "!libs/imng-application-insights-ngrx/**"]
          }
        },
        "test": {
          "builder": "@nrwl/jest:jest",
          "options": {
            "jestConfig": "libs/imng-application-insights-ngrx/jest.config.js",
            "tsConfig": "libs/imng-application-insights-ngrx/tsconfig.spec.json",
            "setupFile": "libs/imng-application-insights-ngrx/src/test-setup.ts",
            "passWithNoTests": true
          }
        }
      },
      "schematics": {
        "@nrwl/angular:component": {
          "styleext": "scss"
        }
      }
    },
    "imng-kendo-data-entry": {
      "projectType": "library",
      "root": "libs/imng-kendo-data-entry",
      "sourceRoot": "libs/imng-kendo-data-entry/src",
      "architect": {
        "build": {
          "builder": "@nrwl/angular:package",
          "options": {
            "tsConfig": "libs/imng-kendo-data-entry/tsconfig.lib.json",
            "project": "libs/imng-kendo-data-entry/ng-package.json"
          },
          "configurations": {
            "production": {
              "tsConfig": "libs/imng-kendo-data-entry/tsconfig.lib.prod.json"
            }
          }
        },
        "lint": {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": {
            "tsConfig": [
              "libs/imng-kendo-data-entry/tsconfig.lib.json",
              "libs/imng-kendo-data-entry/tsconfig.spec.json"
            ],
            "exclude": ["**/node_modules/**", "!libs/imng-kendo-data-entry/**"]
          }
        },
        "test": {
          "builder": "@nrwl/jest:jest",
          "options": {
            "jestConfig": "libs/imng-kendo-data-entry/jest.config.js",
            "tsConfig": "libs/imng-kendo-data-entry/tsconfig.spec.json",
            "setupFile": "libs/imng-kendo-data-entry/src/test-setup.ts",
            "passWithNoTests": true
          }
        }
      },
      "schematics": {
        "@nrwl/angular:component": {
          "styleext": "scss"
        }
      }
    },
    "imng-ngrx-utils": {
      "projectType": "library",
      "root": "libs/imng-ngrx-utils",
      "sourceRoot": "libs/imng-ngrx-utils/src",
      "prefix": "imng",
      "architect": {
        "build": {
          "builder": "@nrwl/angular:package",
          "options": {
            "tsConfig": "libs/imng-ngrx-utils/tsconfig.lib.json",
            "project": "libs/imng-ngrx-utils/ng-package.json"
          },
          "configurations": {
            "production": {
              "tsConfig": "libs/imng-ngrx-utils/tsconfig.lib.prod.json"
            }
          }
        },
        "lint": {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": {
            "tsConfig": ["libs/imng-ngrx-utils/tsconfig.lib.json", "libs/imng-ngrx-utils/tsconfig.spec.json"],
            "exclude": ["**/node_modules/**", "!libs/imng-ngrx-utils/**"]
          }
        },
        "test": {
          "builder": "@nrwl/jest:jest",
          "options": {
            "jestConfig": "libs/imng-ngrx-utils/jest.config.js",
            "tsConfig": "libs/imng-ngrx-utils/tsconfig.spec.json",
            "setupFile": "libs/imng-ngrx-utils/src/test-setup.ts",
            "passWithNoTests": true
          }
        }
      },
      "schematics": {
        "@nrwl/angular:component": {
          "styleext": "scss"
        }
      }
    },
    "imng-nrsrx-client-utils": {
      "projectType": "library",
      "root": "libs/imng-nrsrx-client-utils",
      "sourceRoot": "libs/imng-nrsrx-client-utils/src",
      "prefix": "imng",
      "architect": {
        "build": {
          "builder": "@nrwl/angular:package",
          "options": {
            "tsConfig": "libs/imng-nrsrx-client-utils/tsconfig.lib.json",
            "project": "libs/imng-nrsrx-client-utils/ng-package.json"
          },
          "configurations": {
            "production": {
              "tsConfig": "libs/imng-nrsrx-client-utils/tsconfig.lib.prod.json"
            }
          }
        },
        "lint": {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": {
            "tsConfig": [
              "libs/imng-nrsrx-client-utils/tsconfig.lib.json",
              "libs/imng-nrsrx-client-utils/tsconfig.spec.json"
            ],
            "exclude": ["**/node_modules/**", "!libs/imng-nrsrx-client-utils/**"]
          }
        },
        "test": {
          "builder": "@nrwl/jest:jest",
          "options": {
            "jestConfig": "libs/imng-nrsrx-client-utils/jest.config.js",
            "tsConfig": "libs/imng-nrsrx-client-utils/tsconfig.spec.json",
            "setupFile": "libs/imng-nrsrx-client-utils/src/test-setup.ts",
            "passWithNoTests": true
          }
        }
      },
      "schematics": {
        "@nrwl/angular:component": {
          "styleext": "scss"
        }
      }
    },
    "imng-ngxb-typeahead": {
      "projectType": "library",
      "root": "libs/imng-ngxb-typeahead",
      "sourceRoot": "libs/imng-ngxb-typeahead/src",
      "architect": {
        "build": {
          "builder": "@nrwl/angular:package",
          "options": {
            "tsConfig": "libs/imng-ngxb-typeahead/tsconfig.lib.json",
            "project": "libs/imng-ngxb-typeahead/ng-package.json"
          },
          "configurations": {
            "production": {
              "tsConfig": "libs/imng-ngxb-typeahead/tsconfig.lib.prod.json"
            }
          }
        },
        "lint": {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": {
            "tsConfig": ["libs/imng-ngxb-typeahead/tsconfig.lib.json", "libs/imng-ngxb-typeahead/tsconfig.spec.json"],
            "exclude": ["**/node_modules/**", "!libs/imng-ngxb-typeahead/**"]
          }
        },
        "test": {
          "builder": "@nrwl/jest:jest",
          "options": {
            "jestConfig": "libs/imng-ngxb-typeahead/jest.config.js",
            "tsConfig": "libs/imng-ngxb-typeahead/tsconfig.spec.json",
            "setupFile": "libs/imng-ngxb-typeahead/src/test-setup.ts",
            "passWithNoTests": true
          }
        }
      },
      "schematics": {
        "@nrwl/angular:component": {
          "styleext": "scss"
        }
      }
    },
    "imng-kendo-chart-odata": {
      "projectType": "library",
      "root": "libs/imng-kendo-chart-odata",
      "sourceRoot": "libs/imng-kendo-chart-odata/src",
      "architect": {
        "build": {
          "builder": "@nrwl/angular:package",
          "options": {
            "tsConfig": "libs/imng-kendo-chart-odata/tsconfig.lib.json",
            "project": "libs/imng-kendo-chart-odata/ng-package.json"
          },
          "configurations": {
            "production": {
              "tsConfig": "libs/imng-kendo-chart-odata/tsconfig.lib.prod.json"
            }
          }
        },
        "lint": {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": {
            "tsConfig": [
              "libs/imng-kendo-chart-odata/tsconfig.lib.json",
              "libs/imng-kendo-chart-odata/tsconfig.spec.json"
            ],
            "exclude": ["**/node_modules/**", "!libs/imng-kendo-chart-odata/**"]
          }
        },
        "test": {
          "builder": "@nrwl/jest:jest",
          "options": {
            "jestConfig": "libs/imng-kendo-chart-odata/jest.config.js",
            "tsConfig": "libs/imng-kendo-chart-odata/tsconfig.spec.json",
            "setupFile": "libs/imng-kendo-chart-odata/src/test-setup.ts",
            "passWithNoTests": true
          }
        }
      },
      "schematics": {
        "@nrwl/angular:component": {
          "styleext": "scss"
        }
      }
    },
    "imng-az-func-configr": {
      "projectType": "application",
      "schematics": {
        "@nrwl/angular:component": {
          "style": "scss"
        }
      },
      "root": "apps/imng-az-func-configr",
      "sourceRoot": "apps/imng-az-func-configr/src",
      "prefix": "imng",
      "architect": {
        "build": {
          "builder": "@angular-devkit/build-angular:browser",
          "options": {
            "outputPath": "dist/apps/imng-az-func-configr",
            "index": "apps/imng-az-func-configr/src/index.html",
            "main": "apps/imng-az-func-configr/src/main.ts",
            "polyfills": "apps/imng-az-func-configr/src/polyfills.ts",
            "tsConfig": "apps/imng-az-func-configr/tsconfig.app.json",
            "aot": true,
            "assets": [
              "apps/imng-az-func-configr/src/favicon.ico",
              "apps/imng-az-func-configr/src/assets",
              "apps/imng-az-func-configr/src/manifest.webmanifest"
            ],
            "styles": [
              "./node_modules/bootstrap/dist/css/bootstrap.min.css",
              "./node_modules/jsoneditor/dist/jsoneditor.min.css",
              "apps/imng-az-func-configr/src/styles.scss"
            ],
            "scripts": []
          },
          "configurations": {
            "production": {
              "fileReplacements": [
                {
                  "replace": "apps/imng-az-func-configr/src/environments/environment.ts",
                  "with": "apps/imng-az-func-configr/src/environments/environment.prod.ts"
                }
              ],
              "optimization": true,
              "outputHashing": "all",
              "sourceMap": false,
              "extractCss": true,
              "namedChunks": false,
              "extractLicenses": true,
              "vendorChunk": false,
              "buildOptimizer": true,
              "budgets": [
                {
                  "type": "initial",
                  "maximumWarning": "2mb",
                  "maximumError": "5mb"
                },
                {
                  "type": "anyComponentStyle",
                  "maximumWarning": "6kb",
                  "maximumError": "10kb"
                }
              ],
              "serviceWorker": true,
              "ngswConfigPath": "apps/imng-az-func-configr/ngsw-config.json"
            }
          }
        },
        "serve": {
          "builder": "@angular-devkit/build-angular:dev-server",
          "options": {
            "browserTarget": "imng-az-func-configr:build"
          },
          "configurations": {
            "production": {
              "browserTarget": "imng-az-func-configr:build:production"
            }
          }
        },
        "extract-i18n": {
          "builder": "@angular-devkit/build-angular:extract-i18n",
          "options": {
            "browserTarget": "imng-az-func-configr:build"
          }
        },
        "lint": {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": {
            "tsConfig": ["apps/imng-az-func-configr/tsconfig.app.json", "apps/imng-az-func-configr/tsconfig.spec.json"],
            "exclude": ["**/node_modules/**", "!apps/imng-az-func-configr/**"]
          }
        },
        "test": {
          "builder": "@nrwl/jest:jest",
          "options": {
            "jestConfig": "apps/imng-az-func-configr/jest.config.js",
            "tsConfig": "apps/imng-az-func-configr/tsconfig.spec.json",
            "setupFile": "apps/imng-az-func-configr/src/test-setup.ts",
            "passWithNoTests": true
          }
        }
      }
    },
    "imng-az-func-configr-e2e": {
      "root": "apps/imng-az-func-configr-e2e",
      "sourceRoot": "apps/imng-az-func-configr-e2e/src",
      "projectType": "application",
      "architect": {
        "e2e": {
          "builder": "@nrwl/cypress:cypress",
          "options": {
            "cypressConfig": "apps/imng-az-func-configr-e2e/cypress.json",
            "tsConfig": "apps/imng-az-func-configr-e2e/tsconfig.e2e.json",
            "devServerTarget": "imng-az-func-configr:serve"
          },
          "configurations": {
            "production": {
              "devServerTarget": "imng-az-func-configr:serve:production"
            }
          }
        },
        "lint": {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": {
            "tsConfig": ["apps/imng-az-func-configr-e2e/tsconfig.e2e.json"],
            "exclude": ["**/node_modules/**", "!apps/imng-az-func-configr-e2e/**"]
          }
        }
      }
    },
    "imng-snippets": {
      "projectType": "library",
      "root": "libs/imng-snippets",
      "sourceRoot": "libs/imng-snippets/src",
      "prefix": "imng",
      "architect": {
        "lint": {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": {
            "tsConfig": ["libs/imng-snippets/tsconfig.lib.json", "libs/imng-snippets/tsconfig.spec.json"],
            "exclude": ["**/node_modules/**", "!libs/imng-snippets/**"]
          }
        },
        "test": {
          "builder": "@nrwl/jest:jest",
          "options": {
            "jestConfig": "libs/imng-snippets/jest.config.js",
            "tsConfig": "libs/imng-snippets/tsconfig.spec.json",
            "setupFile": "libs/imng-snippets/src/test-setup.ts",
            "passWithNoTests": true
          }
        }
      },
      "schematics": {}
    },
    "imng-kendo-grid": {
      "projectType": "library",
      "root": "libs/imng-kendo-grid",
      "sourceRoot": "libs/imng-kendo-grid/src",
      "prefix": "imng",
      "architect": {
        "build": {
          "builder": "@nrwl/angular:package",
          "options": {
            "tsConfig": "libs/imng-kendo-grid/tsconfig.lib.json",
            "project": "libs/imng-kendo-grid/ng-package.json"
          },
          "configurations": {
            "production": {
              "tsConfig": "libs/imng-kendo-grid/tsconfig.lib.prod.json"
            }
          }
        },
        "lint": {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": {
            "tsConfig": ["libs/imng-kendo-grid/tsconfig.lib.json", "libs/imng-kendo-grid/tsconfig.spec.json"],
            "exclude": ["**/node_modules/**", "!libs/imng-kendo-grid/**"]
          }
        },
        "test": {
          "builder": "@nrwl/jest:jest",
          "options": {
            "jestConfig": "libs/imng-kendo-grid/jest.config.js",
            "tsConfig": "libs/imng-kendo-grid/tsconfig.spec.json",
            "setupFile": "libs/imng-kendo-grid/src/test-setup.ts",
            "passWithNoTests": true
          }
        }
      },
      "schematics": {
        "@nrwl/angular:component": {
          "styleext": "scss"
        }
      }
    },
    "nurse-cron": {
      "projectType": "application",
      "schematics": {
        "@nrwl/angular:component": {
          "style": "scss"
        }
      },
      "root": "apps/nurse-cron",
      "sourceRoot": "apps/nurse-cron/src",
      "prefix": "nrcrn",
      "architect": {
        "build": {
          "builder": "@angular-devkit/build-angular:browser",
          "options": {
            "outputPath": "dist/apps/nurse-cron",
            "index": "apps/nurse-cron/src/index.html",
            "main": "apps/nurse-cron/src/main.ts",
            "polyfills": "apps/nurse-cron/src/polyfills.ts",
            "tsConfig": "apps/nurse-cron/tsconfig.app.json",
            "aot": true,
            "assets": [
              "apps/nurse-cron/src/favicon.ico",
              "apps/nurse-cron/src/assets",
              {
                "glob": "*",
                "input": "libs/imng-auth0-oidc/src/assets",
                "output": "/"
              }
            ],
            "styles": ["apps/nurse-cron/src/styles.scss"],
            "scripts": []
          },
          "configurations": {
            "production": {
              "fileReplacements": [
                {
                  "replace": "apps/nurse-cron/src/environments/environment.ts",
                  "with": "apps/nurse-cron/src/environments/environment.prod.ts"
                }
              ],
              "optimization": true,
              "outputHashing": "all",
              "sourceMap": false,
              "extractCss": true,
              "namedChunks": false,
              "extractLicenses": true,
              "vendorChunk": false,
              "buildOptimizer": true,
              "budgets": [
                {
                  "type": "initial",
                  "maximumWarning": "2mb",
                  "maximumError": "5mb"
                },
                {
                  "type": "anyComponentStyle",
                  "maximumWarning": "6kb",
                  "maximumError": "10kb"
                }
              ]
            }
          }
        },
        "serve": {
          "builder": "@angular-devkit/build-angular:dev-server",
          "options": {
            "browserTarget": "nurse-cron:build",
            "proxyConfig": "apps/nurse-cron/proxy.conf.json"
          },
          "configurations": {
            "production": {
              "browserTarget": "nurse-cron:build:production",
              "proxyConfig": "apps/nurse-cron/proxy.conf.json"
            }
          }
        },
        "extract-i18n": {
          "builder": "@angular-devkit/build-angular:extract-i18n",
          "options": {
            "browserTarget": "nurse-cron:build"
          }
        },
        "lint": {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": {
            "tsConfig": ["apps/nurse-cron/tsconfig.app.json", "apps/nurse-cron/tsconfig.spec.json"],
            "exclude": ["**/node_modules/**", "!apps/nurse-cron/**"]
          }
        },
        "test": {
          "builder": "@nrwl/jest:jest",
          "options": {
            "jestConfig": "apps/nurse-cron/jest.config.js",
            "tsConfig": "apps/nurse-cron/tsconfig.spec.json",
            "passWithNoTests": true,
            "setupFile": "apps/nurse-cron/src/test-setup.ts"
          }
        }
      }
    },
    "nurse-cron-e2e": {
      "root": "apps/nurse-cron-e2e",
      "sourceRoot": "apps/nurse-cron-e2e/src",
      "projectType": "application",
      "architect": {
        "e2e": {
          "builder": "@nrwl/cypress:cypress",
          "options": {
            "cypressConfig": "apps/nurse-cron-e2e/cypress.json",
            "tsConfig": "apps/nurse-cron-e2e/tsconfig.e2e.json",
            "devServerTarget": "nurse-cron:serve"
          },
          "configurations": {
            "production": {
              "devServerTarget": "nurse-cron:serve:production"
            }
          }
        },
        "lint": {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": {
            "tsConfig": ["apps/nurse-cron-e2e/tsconfig.e2e.json"],
            "exclude": ["**/node_modules/**", "apps/nurse-cron-e2e/src/plugins/index.js", "!apps/nurse-cron-e2e/**"]
          }
        }
      }
    }
  },
  "cli": {
    "warnings": {
      "typescriptMismatch": false,
      "versionMismatch": false
    },
    "defaultCollection": "@nrwl/angular",
    "analytics": "f4c251a3-a9e0-433e-b1f3-2c10661975ea"
  },
  "schematics": {
    "@nrwl/angular:application": {
      "unitTestRunner": "jest",
      "e2eTestRunner": "cypress"
    },
    "@nrwl/angular:library": {
      "unitTestRunner": "jest"
    }
  },
  "defaultProject": "nurse-cron"
}
