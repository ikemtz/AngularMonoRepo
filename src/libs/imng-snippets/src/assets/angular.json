{
  "ComponentMethodsTypeAhead": {
    "prefix": "imng-component-methods-type-ahead",
    "body": [
      "  public onTypeAheadSelected(match: ImngMatchSelectedEvent<I${1:__ENTITY_NAME__PASCAL_CASE__}>) {",
      "  }",
      "  public onTypeAheadCleared(noMatches: boolean) {",
      "  }"
    ],
    "description": "ComponentMethodsTypeAhead"
  },
  "ServicesDataEntry": {
    "prefix": "imng-services-data-entry",
    "body": [
      "import { HttpClient } from '@angular/common/http';",
      "import { Injectable } from '@angular/core';",
      "import { NrsrxBaseApiClientService } from 'imng-nrsrx-client-utils';",
      "@Injectable({",
      "  providedIn: 'root',",
      "})",
      "export class ${1:__ENTITY_NAME__PASCAL_CASE__}Service extends NrsrxBaseApiClientService<I${1:__ENTITY_NAME__PASCAL_CASE__}> {",
      "  url: string = environment.endPoints.${2:__NGRX_FEATURE_KEY__CAMEL_CASE__}s.${1:__ENTITY_NAME__PASCAL_CASE__};",
      "  constructor(http: HttpClient) {",
      "    super(http);",
      "  }",
      "}"
    ],
    "description": "ServicesDataEntry"
  }
}
