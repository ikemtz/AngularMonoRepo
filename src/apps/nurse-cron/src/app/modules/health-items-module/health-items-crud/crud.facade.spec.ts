import { NgModule } from '@angular/core';
import { TestBed } from '@angular/core/testing';
import { EffectsModule } from '@ngrx/effects';
import { StoreModule, Store } from '@ngrx/store';
import { NxModule } from '@nrwl/angular';
import { readFirst } from '@nrwl/angular/testing';
import {
  testAddSetAndClearCurrentEntity,
  testEditSetAndClearCurrentEntity,
  testSaveCurrentEntity,
  testUpdateCurrentEntity,
} from 'imng-kendo-data-entry/testing';
import { HttpClient } from '@angular/common/http';
import { HttpClientTestingModule } from '@angular/common/http/testing';
import { environment } from '@env';

import { HealthItemEffects } from '../+state/health-item.effects';
import {
  HealthItemsPartialState,
  initialState,
  reducer as healthItemReducer,
  HEALTH_ITEMS_FEATURE_KEY
} from '../+state/health-item.reducer';
import { HealthItemCrudFacade } from './crud.facade';
import { HealthItemApiService } from './api.service';

interface TestSchema {
  [HEALTH_ITEMS_FEATURE_KEY]: HealthItemsPartialState;
}

export const createHealthItem = () => 
  <IHealthItem>{
    [HealthItemProperties.ID]: 'ID',
    [HealthItemProperties.NAME]: 'NAME',
    [HealthItemProperties.IS_ENABLED]: true,
    };

describe('HealthItemCrudFacade', () => {
  let facade: HealthItemCrudFacade;
  let store: Store<TestSchema>;

  beforeEach(() => {});

  describe('used in NgModule', () => {
    beforeEach(() => {
      @NgModule({
        imports: [
          StoreModule.forFeature(HEALTH_ITEMS_FEATURE_KEY, healthItemReducer, { initialState }),
          EffectsModule.forFeature([HealthItemEffects]),
          HttpClientTestingModule,
        ],
        providers: [HealthItemCrudFacade, HealthItemApiService],
      })
      class CustomFeatureModule {}

      @NgModule({
        imports: [
          NxModule.forRoot(),
          StoreModule.forRoot({}, { runtimeChecks: environment.runtimeChecks }),
          EffectsModule.forRoot([]),
          CustomFeatureModule,
        ],
      })
      class RootModule {}
      TestBed.configureTestingModule({ imports: [RootModule] });

      store = TestBed.inject(Store);
      facade = TestBed.inject(HealthItemCrudFacade);
    });

    it('clearCurrentEntity() should set currentHealthItem to null', async done => {
      try {
        let isNewActive = await readFirst(facade.isNewActive$);
        expect(isNewActive).toBeFalsy();

        facade.clearCurrentEntity();
        isNewActive = await readFirst(facade.isNewActive$);

        expect(isNewActive).toBeFalsy();

        done();
      } catch (err) {
        done.fail(err);
      }
    });

    it('saveNewEntity() should save new Entity', async done => {
      try {
        let isNewActive = await readFirst(facade.isNewActive$);
        expect(isNewActive).toBeFalsy();

        facade.saveNewEntity(createHealthItem());
        isNewActive = await readFirst(facade.isNewActive$);
        expect(isNewActive).toBeFalsy();

        done();
      } catch (err) {
        done.fail(err);
      }
    });

    it('New Entity Set And Clear CurrentEntity', async done =>
      testAddSetAndClearCurrentEntity<HealthItemCrudFacade>(done, facade));
    it('Existing Entity Set And Clear CurrentEntity', async done =>
      testEditSetAndClearCurrentEntity<HealthItemCrudFacade>(done, facade));
    it('Save CurrentEntity', async done =>
      testSaveCurrentEntity<HealthItemCrudFacade>(done, facade, TestBed.inject(HttpClient)));
    it('Update CurrentEntity', async done =>
      testUpdateCurrentEntity<HealthItemCrudFacade>(done, facade, TestBed.inject(HttpClient)));
  });
});
