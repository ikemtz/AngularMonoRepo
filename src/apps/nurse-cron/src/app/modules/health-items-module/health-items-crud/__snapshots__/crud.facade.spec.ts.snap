// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HealthItemCrudFacade used in NgModule clearCurrentEntity() should set currentHealthItem to null 1`] = `
Object {
  "healthItems": Object {
    "currentHealthItem": null,
    "gridData": Object {
      "data": Array [],
      "total": 0,
    },
    "gridODataState": Object {},
    "gridPagerSettings": false,
    "loading": false,
  },
}
`;
